src/app/ui/auto-complete/select/select.component.ts

```diff
@@ -121,6 +124,10 @@ export class SelectComponent implements AfterViewInit, OnDestroy {
     @Input()
     public labelField = "text";
 
+    // When onChange is triggered and this flag is false we know that change comes during items/options initializing process
+    // and we should not propagate the change (if we would, we would always have form dirty flag set)
+    protected shouldTriggerChange = false;
```

Comment:
Problem here is particularly connect to selectize component. updateOptions() method is creating items and for creating each item it is triggering onChange() event which is correct behavior. But problem here is when items/options are initialized, in that case creating each item triggers onChange() event, and form control which is using AutoComplete component, will always be dirty. Changing items/options externally should not affect on form dirtiness, only user interactions with the component. I did not find better way to resolve this problem because add/create item function (selectize api) does not have "silent" parameter or similar.